internal/services/newrelic/new_relic_monitor_resource.go

```diff
@@ -205,6 +215,21 @@ func (r NewRelicMonitorResource) Arguments() map[string]*pluginsdk.Schema {
 			ValidateFunc: validation.StringIsNotEmpty,
 		},
 
+		"monitored_subscription": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"subscription_id": {
```

Comment:
This is not flattened because there are other properties in the [monitoredSubscription](https://github.com/Azure/azure-rest-api-specs/blob/4aad50a36767f7c36673f2c7982bb4055dbf5ed4/specification/newrelic/resource-manager/NewRelic.Observability/stable/2024-03-01/NewRelic.json#L2832-L2865) but not supported by API for now and it might be supported in the future.